### PR TITLE
[BUGFIX] fix processing of page attribute varnish_cache in PostProces…

### PR DESCRIPTION
**Pull request by crazymushrooms**
…sOutputHook; add functional tests

### CODE REVIEW

File: Tests/Functional/FunctionalTests.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<phpunit
```
Comment (tomasnorre) [inline Tests/Functional/FunctionalTests.xml]: Why is a custom FunctionalTests.xml needed? Normally the one from Nimut-testingframework works perfectly well.

File: .travis.yml [hunk 2]
```diff
@@ -6,9 +6,12 @@ php:
   - 7.2
 
 env:
-  - TYPO3_VERSION=^7.6 COVERAGE=0
-  - TYPO3_VERSION=^8.7 COVERAGE=0
-  - TYPO3_VERSION=dev-master COVERAGE=0
+  - TYPO3_VERSION=^7.6 COVERAGE=0 FUNCTIONAL_TEST=0
+  - TYPO3_VERSION=^8.7 COVERAGE=0 FUNCTIONAL_TEST=1 typo3DatabaseName=test typo3DatabaseHost=127.0.0.1 typo3DatabaseUsername=travis typo3DatabasePassword=root
```
Comment (tomasnorre) [inline .travis.yml]: I would have the DB information in the command where the tests are executed, to keep the matrix cleaner.

File: Tests/Functional/System/SetCorrectHeadersTest.php [hunk 3]
```diff
@@ -0,0 +1,182 @@
+<?php
+
```
Comment (tomasnorre) [inline Tests/Functional/System/SetCorrectHeadersTest.php]: Remove empty lines please.

File: Tests/Functional/System/SetCorrectHeadersTest.php [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        parent::setUp();
+
+        try {
+            $this->importDataSet($this->fixturePath . '/DataSet.xml');
+        } catch (\Exception $e) {
+            throw new \Error($e);
+        }
+    }
+
+
```
Comment (tomasnorre) [inline Tests/Functional/System/SetCorrectHeadersTest.php]: Remove empty line

File: composer.json [hunk 5]
```diff
@@ -49,6 +48,10 @@
         "test:unit": [
             "[ -e .Build/bin/phpunit ] || composer update",
             "TYPO3_PATH_ROOT=.Build/Web .Build/bin/phpunit -c .Build/vendor/nimut/testing-framework/res/Configuration/UnitTests.xml Tests/Unit/ --debug --verbose"
+        ],
+        "test:functional": [
+            "[ -e .Build/bin/phpunit ] || composer update",
+            "TYPO3_PATH_ROOT=$PWD/.Build/Web typo3DatabaseName=\"test\" typo3DatabaseHost=\"127.0.0.1\" typo3DatabaseUsername=\"root\" typo3DatabasePassword=\"root\" .Build/bin/phpunit -c Tests/Functional/FunctionalTests.xml Tests/Functional"
```
Comment (tomasnorre) [inline composer.json]: You can use single quotes, then you don't need to escape them.

File: .travis.yml [hunk 6]
```diff
@@ -6,9 +6,12 @@ php:
   - 7.2
 
 env:
-  - TYPO3_VERSION=^7.6 COVERAGE=0
-  - TYPO3_VERSION=^8.7 COVERAGE=0
-  - TYPO3_VERSION=dev-master COVERAGE=0
+  - TYPO3_VERSION=^7.6 COVERAGE=0 FUNCTIONAL_TEST=0
```
Comment (tomasnorre) [inline .travis.yml:10]: The 7.6 tests will not work anymore with the composer json adjustments.

File: Tests/Functional/FunctionalTests.xml (hunk 1 above)
Comment (crazymushrooms) [reply Tests/Functional/FunctionalTests.xml]: ok, I'll try it with nimbus configuration.

File: .travis.yml (hunk 2 above)
Comment (crazymushrooms) [reply .travis.yml]: ok

File: Tests/Functional/System/SetCorrectHeadersTest.php (hunk 3 above)
Comment (crazymushrooms) [reply Tests/Functional/System/SetCorrectHeadersTest.php]: ok

File: composer.json (hunk 5 above)
Comment (crazymushrooms) [reply composer.json]: right :)

**Review by TomSchenk — APPROVED** [review]

**Review by tomasnorre — APPROVED** [review]

**Pull request merged**